src/ILCompiler/src/Program.cs

```diff
@@ -178,10 +175,37 @@ private int Run(string[] args)
                 SharedGenericsMode.CanonicalReferenceTypes : SharedGenericsMode.Disabled;
 
             var typeSystemContext = new CompilerTypeSystemContext(new TargetDetails(_targetArchitecture, _targetOS, TargetAbi.CoreRT), genericsMode);
-            typeSystemContext.InputFilePaths = _inputFilePaths;
+
+            //
+            // TODO: To support our pre-compiled test tree, allow input files that aren't managed assemblies since
+            // some tests contain a mixture of both managed and native binaries.
+            //
+            // See: https://github.com/dotnet/corert/issues/2785
+            //
+            // When we undo this this hack, replace this foreach with
+            //  typeSystemContext.InputFilePaths = _inputFilePaths;
+            //
+            Dictionary<string, string> inputFilePaths = new Dictionary<string, string>();
+            foreach (var inputFile in _inputFilePaths)
+            {
+                try
+                {
+                    var module = typeSystemContext.GetModuleFromPath(inputFile.Value);
+                    inputFilePaths.Add(inputFile.Key, inputFile.Value);
+                }
+                catch (Exception)
```

Comment:
Can we catch something more specific? I want to make sure "file doesn't exist", "can't read the file", etc. to turn into these silent failures.